frontend/src/scenes/notebooks/Nodes/NodeWrapper.tsx

```diff
@@ -80,62 +87,134 @@ export function NodeWrapper({
+                                        size="small"
+                                        status="primary-alt"
+                                        className="flex-1"
+                                        icon={<IconDragHandle className="cursor-move text-base shrink-0" />}
+                                    >
+                                        <span className="flex-1 cursor-pointer">{title}</span>
+                                    </LemonButton>
+
+                                    {parsedHref && <LemonButton size="small" icon={<IconLink />} to={parsedHref} />}
+
+                                    <LemonButton
+                                        onClick={() => setExpanded(!expanded)}
+                                        size="small"
+                                        icon={expanded ? <IconUnfoldLess /> : <IconUnfoldMore />}
+                                    />
+                                </div>
+                                <div
+                                    ref={contentRef}
+                                    className={clsx(
+                                        'NotebookNode__content flex flex-col relative z-0 overflow-hidden',
+                                        expanded && resizeable && 'resize-y'
                                     )}
+                                    // eslint-disable-next-line react/forbid-dom-props
+                                    style={expanded && resizeable ? { height, minHeight } : {}}
+                                    onClick={!expanded ? () => setExpanded(true) : undefined}
+                                    onMouseDown={onResizeStart}
+                                >
+                                    {children}
                                 </div>
-                            </div>
-                            <div
-                                ref={contentRef}
-                                className={clsx(
-                                    'flex flex-col relative z-0 overflow-hidden min-h-40',
-                                    resizeable && 'resize-y'
-                                )}
-                                // eslint-disable-next-line react/forbid-dom-props
-                                style={{ height }}
-                                onMouseDown={onResizeStart}
-                            >
-                                {children}
-                            </div>
-                        </>
-                    )}
-                </ErrorBoundary>
-            </NodeViewWrapper>
+                            </>
+                        )}
+                    </ErrorBoundary>
+                </NodeViewWrapper>
+            </BindLogic>
         </NotebookNodeContext.Provider>
     )
 }
+
+export type CreatePostHogWidgetNodeOptions = NodeWrapperProps & {
+    nodeType: NotebookNodeType
+    Component: (props: NodeViewProps) => JSX.Element
+    pasteOptions?: {
```

Comment:
Currently this will clash with the idea of a node having multiple paste rules introduced in https://github.com/PostHog/posthog/pull/16805. I can fix that up once this is merged